### PR TITLE
Pivot to master

### PR DESCRIPTION
**Pull request by greenblitz4590**

### CODE REVIEW

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/Chassis/SwerveChassis.java [hunk 1]
```diff
@@ -110,7 +110,7 @@ public void periodic() {
 		updateInputs(ChassisInputs);
 
 		Logger.recordOutput("DriveTrain/RobotPose", getRobotPose());
-		Logger.recordOutput("DriveTrain/ModuleStates", getSwerveModuleStates());
+//		Logger.recordOutput("DriveTrain/ModuleStates", getSwerveModuleStates());
```
Comment (tnrz135) [inline src/main/java/edu/greenblitz/robotName/subsystems/swerve/Chassis/SwerveChassis.java:113]: .-.

File: src/main/java/edu/greenblitz/robotName/utils/GBMath.java [hunk 2]
```diff
@@ -15,6 +18,10 @@ public class GBMath {
 	public static double absoluteModulo(double x, double y) {
 		return ((x % y) + y) % y;
 	}
-	
-	
+
+	public static double reverseAngle(double angleInRadians){
+		return angleInRadians == 0 ? 0 : angleInRadians > 0 ? -Math.PI * 2 + angleInRadians : angleInRadians + Math.PI * 2;
```
Comment (tnrz135) [inline src/main/java/edu/greenblitz/robotName/utils/GBMath.java:23]: This function is not that clean, especially when the function name is not perfect.

**Review by tnrz135 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/subsystems/Shooter/Pivot/PivotFactory.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import edu.greenblitz.robotName.subsystems.Shooter.Pivot.FalconPivot.FalconPivot;
+import edu.greenblitz.robotName.subsystems.Shooter.Pivot.SimulationPivot.SimulationPivot;
+
+
+public class PivotFactory {
+    public static IPivot create() {
+        return switch (RobotConstants.ROBOT_TYPE) {
+            case ROBOT_NAME -> new FalconPivot();
+            case SIMULATION -> new SimulationPivot();
+            default -> new ReplayPivot();
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/Shooter/Pivot/PivotFactory.java:13]: just make a replay case, and make simulation default

File: src/main/java/edu/greenblitz/robotName/subsystems/Shooter/Pivot/SimulationPivot/SimulationPivot.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+import static edu.greenblitz.robotName.subsystems.Shooter.Pivot.SimulationPivot.SimulationPivotConstants.*;
+
+
+public class SimulationPivot implements IPivot {
+
+    SingleJointedArmSim pivotSimulation;
+
+    private double appliedVoltage;
+
+    private PIDController controller = SIM_PID.getPIDController();
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/Shooter/Pivot/SimulationPivot/SimulationPivot.java:24]: initialize it in the constructor

File: src/main/java/edu/greenblitz/robotName/subsystems/Shooter/Pivot/SimulationPivot/SimulationPivot.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+
+    @Override
+    public void setPower(double power) {
+        setVoltage(power * BATTERY_VOLTAGE);
+    }
+
+    @Override
+    public void setVoltage(double voltage) {
+        appliedVoltage = MathUtil.clamp(voltage, -RobotConstants.SimulationConstants.MAX_MOTOR_VOLTAGE, RobotConstants.SimulationConstants.MAX_MOTOR_VOLTAGE);
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/Shooter/Pivot/SimulationPivot/SimulationPivot.java:51]: just make a MIN_MOTOR_VOLTAGE

File: src/main/java/edu/greenblitz/robotName/subsystems/Shooter/Pivot/SimulationPivot/SimulationPivotConstants.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+
+public class SimulationPivotConstants {
+
+
+    public static final int NUMBER_OF_MOTORS = 1;
+
+    public static final double GEAR_RATIO = 1 / RELATIVE_POSITION_CONVERSION_FACTOR;
+
+    public static final PIDObject SIM_PID = new PIDObject().withKp(0.8).withKd(0.3).withMaxPower(1);
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/Shooter/Pivot/SimulationPivot/SimulationPivotConstants.java:15]: DID SOMEONE SAY SIM????

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]
great job!

File: src/main/java/edu/greenblitz/robotName/subsystems/Shooter/Pivot/SimulationPivot/SimulationPivot.java (hunk 5 above)
Comment (greenblitz4590) [reply src/main/java/edu/greenblitz/robotName/subsystems/Shooter/Pivot/SimulationPivot/SimulationPivot.java:52]: when you merge shooter change it

**Review by bossyshrimp68 — APPROVED** [review]

File: src/main/java/edu/greenblitz/robotName/commands/shooter/pivot/movePivotToAngle/MovePivotToAngle.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
+package edu.greenblitz.robotName.commands.shooter.pivot.movePivotToAngle;
+
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.ConditionalCommand;
+
+import java.util.function.BooleanSupplier;
+
+public class MovePivotToAngle extends ConditionalCommand {
+	
+	public MovePivotToAngle(double goalAngle, boolean isReal) {
```
Comment (itamaroryan) [inline src/main/java/edu/greenblitz/robotName/commands/shooter/pivot/movePivotToAngle/MovePivotToAngle.java:10]: Please change this to `isSimulation` or something like that.

**Review by itamaroryan — CHANGES_REQUESTED** [review]
Please Ctrl+Alt+L before pushing the final version of the code <3

**Review by itamaroryan — APPROVED** [review]

**Review by YoniKiriaty — APPROVED** [review]
Amireeeee

**Pull request merged**